### ISSUE TITLE
swagger-ui.html can't be proxied by another path - js error occurs

### ISSUE THREAD

**Issue by bitsal** [springfox/springfox#2940]
Please take the time to search the repository, if your question has already been asked or answered.

- Version 2.9.2

**Steps to reproduce:**
1. There is a spring boot application -> REST API, e.g. it runs on `localhost:1001/api`.
Run a proxy, e.g. nginx on `localhost:1002` with proxy configuration
/api-docs -> localhost:1001/api/swagger-ui.html
/webjars/springfox-swagger-ui -> localhost:1001/api$request_uri
/swagger-resources -> localhost:1001/api$request_uri
2. Open localhost:1002/api-docs

**Actual result:**
All resources are loaded. JS error occurs.
```
Uncaught (in promise) TypeError: Cannot read property '1' of null
    at e (springfox.js?v=2.9.2:1)
    at springfox.js?v=2.9.2:1
    at x (springfox.js?v=2.9.2:1)
    at Generator._invoke (springfox.js?v=2.9.2:1)
    at Generator.t.(anonymous function) [as next] (http://localhost:9080/webjars/springfox-swagger-ui/springfox.js?v=2.9.2:1:50487)
    at e (springfox.js?v=2.9.2:1)
    at springfox.js?v=2.9.2:1
    at new Promise (<anonymous>)
    at springfox.js?v=2.9.2:1
    at window.onload (springfox.js?v=2.9.2:1)
```

**Expected result:**
It should work as expected.

**Noted in JS code:**
`return/(.*)\/swagger-ui.html.*/.exec(window.location.href)[1]` - but window.location.href is a proxy href.

Please fix it or help me to find a workaround until you fix it.

**Comment by bitsal** [comment]
No activity from repo owners, please explain why it won't be fixed
Or probably it is fixed in newer versions

**Comment by halungge** [comment]
I have the exaxt same problem. Please provide a workaround.

**Comment by halungge** [comment]
as a workaround: you have to make the proxy href end with `swagger-ui.html` for the code line above to work. Change your nginx configuration to contain something like: 
`
location ^~ /api-docs/swagger-ui.html {
      proxy_pass  localhost:1001/api/swagger-ui.html;
}
`

**Comment by dilipkrish** [comment]
@halungge This should be fixed and would work much better in 3.0.0. 

NOTE: the url has changed from /swagger-ui.html to having everything related to swagger ui under `/swagger-ui/**` and what used to be swagger-ui.html is now index.html under that root

**Comment by TheSawIsTheLaw** [comment]
Still having same problem with 
```
 Uncaught (in promise) TypeError: Cannot read properties of null (reading '1')
    at springfox.js:139
    at s (runtime.js:45) 
```

String 139 still contains:
```js
const urlMatches = /(.*)\/swagger-ui(\/index.html.*)*/.exec(window.location.href);
```

Springfox version: 3.0.0
I've tried my best to solve this problem but after a week I have no more strength to fight with this one :)

Problem shows with this nginx configuration part:
```nginx
upstream servers {
    server service1:8080 fail_timeout=50s max_fails=5 weight=2;
    server service2:8080 fail_timeout=50s max_fails=5;
    server service2:8080 fail_timeout=50s max_fails=5;
}

server {
    ...
    location /api/ {
        proxy_no_cache 1;
        proxy_pass http://servers/swagger-ui/;
    }
}
```

And, of course, when i using something really terrible like:
```nginx
location /api { #/ {
        proxy_no_cache 1;
        rewrite ^/api$ http://localhost:8181/swagger-ui/ redirect;
    }
```
It works.
So, I have no idea how to solve this problem. My seminarian will kill me, i guess...
